choreolib/src/main/java/choreo/auto/AutoFactory.java

```diff
@@ -125,13 +121,13 @@ HashMap<String, Command> getBindings() {
     }
   }
 
-  private final TrajectoryCache trajectoryCache = new TrajectoryCache();
-  private final Supplier<Pose2d> poseSupplier;
-  private final BiConsumer<Pose2d, ? extends TrajectorySample<?>> controller;
-  private final BooleanSupplier mirrorTrajectory;
-  private final Subsystem driveSubsystem;
-  private final AutoBindings bindings = new AutoBindings();
-  private final Optional<TrajectoryLogger<? extends TrajectorySample<?>>> trajectoryLogger;
+  final TrajectoryCache trajectoryCache = new TrajectoryCache();
+  final Supplier<Pose2d> poseSupplier;
+  final BiConsumer<Pose2d, ? extends TrajectorySample<?>> controller;
+  final BooleanSupplier mirrorTrajectory;
+  final Subsystem driveSubsystem;
+  final AutoBindings bindings = new AutoBindings();
+  final Optional<TrajectoryLogger<? extends TrajectorySample<?>>> trajectoryLogger;
```

Comment:
this feels off to me; I don't get the benefit from breaking encapsulation like this

Reply:
I could pass them in as dependencies instead of passing in the AutoFactory itself. Overall though, it still doesn't break encapsulation from the library side(and was the only way I could do this)